lib/project-data.ts

```diff
@@ -87,6 +89,50 @@ export class ProjectData implements IProjectData {
 		this.$errors.fail("No project found at or above '%s' and neither was a --path specified.", projectDir || this.$options.path || currentDir);
 	}
 
+	public getAppResourcesDirectoryPath(projectDir?: string): string {
+		if (!projectDir) {
+			projectDir = this.projectDir;
+		}
+
+		const configNS = this.getNsConfig(projectDir);
+		let absoluteAppResourcesDirPath: string;
+
+		if (configNS && configNS[constants.CONFIG_NS_APP_RESOURCES_ENTRY]) {
+			const appResourcesDirPath = configNS[constants.CONFIG_NS_APP_RESOURCES_ENTRY];
+
+			absoluteAppResourcesDirPath = path.resolve(projectDir, appResourcesDirPath);
+		}
+
+		return absoluteAppResourcesDirPath || path.join(this.getAppDirectoryPath(projectDir), constants.APP_RESOURCES_FOLDER_NAME);
+	}
+
+	public getAppDirectoryPath(projectDir?: string): string {
```

Comment:
This method seems to be very similar to the one above (`getAppResourcesDirectoryPath`). Do you think it would be a good idea to make a common method that accepts the property name as an argument? Or maybe have a method that parses the config file and validates it?